MiniApp/Classes/core/Downloader/MiniAppClient.swift

```diff
@@ -145,7 +161,22 @@ internal class MiniAppClient: NSObject, URLSessionDownloadDelegate {
             delegate?.downloadFileTaskCompleted(url: "", error: NSError.downloadingFailed())
             return
         }
-        delegate?.fileDownloaded(at: location, downloadedURL: destinationURL)
+        let ids = Self.miniAppVersionIdFromZipUrl(url: destinationURL)
+        #if RMA_SDK_SIGNATURE
+            let requireMiniAppSignatureVerification = environment.requireMiniAppSignatureVerification
+            if let versionId = ids.1, let data = try? Data(contentsOf: location) {
+                verifySignature(version: versionId, signature: signatures[versionId]?.1 ?? "", keyId: signatures[versionId]?.0 ?? "", data: data) { [weak self] isVerified in
+                    if !isVerified { MiniAppAnalytics.sendAnalytics(event: .signatureFailure, miniAppId: ids.0, miniAppVersion: versionId) }
+                    self?.delegate?.fileDownloaded(at: location, downloadedURL: destinationURL, signatureChecked: isVerified || !requireMiniAppSignatureVerification)
+                }
+            } else {
+                MiniAppAnalytics.sendAnalytics(event: .signatureFailure, miniAppId: ids.0, miniAppVersion: ids.1)
+                delegate?.fileDownloaded(at: location, downloadedURL: destinationURL, signatureChecked: !requireMiniAppSignatureVerification)
+            }
+        #else
+            MiniAppAnalytics.sendAnalytics(event: .signatureFailure, miniAppId: ids.0, miniAppVersion: ids.1)
+            delegate?.fileDownloaded(at: location, downloadedURL: destinationURL)
+        #endif
```

Comment:
Do you think we can movubg this to a separate method would be good idea?

Reply:
yeah, makes things cleaner